### ISSUE TITLE
Simple Dandisets failing to validate within 20 seconds

### ISSUE THREAD

**Issue by jwodder** [dandi/dandi-archive#548]
As seen in #538 and [here](https://github.com/dandi/dandischema/pull/91/checks?check_run_id=3778328101#step:12:5835), dandi-api (specifically, dandi-staging and Dockerized environments; not yet tested in production) occasionally fails to validate simple Dandisets containing only a few text files within the 20 seconds that dandi-cli waits for validity.

The following script can reproduce the error, but it's not guaranteed, so you may have to run it a couple times:

```python
import os
from pathlib import Path
from tempfile import TemporaryDirectory

from dandi.consts import dandiset_metadata_file
from dandi.dandiapi import DandiAPIClient
from dandi.upload import upload

with DandiAPIClient.for_dandi_instance(
    "dandi-staging", token=os.environ["DANDI_API_KEY"]
) as client:
    d = client.create_dandiset(
        "Text Dandiset",
        {
            "schemaKey": "Dandiset",
            "name": "Text Dandiset",
            "description": "A test text Dandiset",
            "contributor": [
                {
                    "schemaKey": "Person",
                    "name": "Wodder, John",
                    "roleName": ["dcite:Author", "dcite:ContactPerson"],
                }
            ],
            "license": ["spdx:CC0-1.0"],
            "manifestLocation": ["https://github.com/dandi/dandi-cli"],
        },
    )
    dandiset_id = d.identifier
    print("DANDISET ID:", dandiset_id)
    with TemporaryDirectory() as tmpdir:
        dspath = Path(tmpdir)
        (dspath / dandiset_metadata_file).write_text(f"identifier: '{dandiset_id}'\n")
        (dspath / "file.txt").write_text("This is test text.\n")
        (dspath / "subdir1").mkdir()
        (dspath / "subdir1" / "apple.txt").write_text("Apple\n")
        (dspath / "subdir2").mkdir()
        (dspath / "subdir2" / "banana.txt").write_text("Banana\n")
        (dspath / "subdir2" / "coconut.txt").write_text("Coconut\n")
        upload(
            paths=None,
            dandiset_path=dspath,
            dandi_instance="dandi-staging",
            # devel_debug=True,
            allow_any_path=True,
            validation="skip",
        )
    d.wait_until_valid()
    v = d.publish().version.identifier
    client.delete(f"{d.api_path}versions/{v}/")
    d.delete()
```

**Comment by dchiquito** [comment]
I added a setting `DJANGO_DANDI_VALIDATION_JOB_INTERVAL` which sets how many seconds to wait between checks for validation jobs. In staging, this is set to 60 seconds, so it makes sense that you would see failures most of the time when running against staging.

I set that value in the dandi-cli docker-compose environment to 5 seconds in https://github.com/dandi/dandi-cli/pull/778, so this should not be happening in local runs or in CI. :eyes:

**Comment by dchiquito** [comment]
I've tried modifying the script to point to `dandi-api-local-docker-tests` and it's working for me, both against my normal `dandi-api` dev environment and the environment from `dandi-cli/dandi/tests/data/dandiarchive-docker`.

@jwodder have you reproduced this with a local dockerized environment, or just the pytest managed environment?

**Comment by jwodder** [comment]
@dchiquito The second link in my OP points to a failure in a Docker environment experienced by the dandischema CI.  I seem to recall seeing this with dandi-cli tests, but I can't find any occurrences right now.  (However, the latest dandi-cli tests are occasionally failing at the same point with the message "Digest is missing dandi-etag or sha256 keys."  Not sure if that's a separate issue or not.)

**Comment by dchiquito** [comment]
I'm unable to reproduce the dandi-cli test failures with repeated `tox` runs.

Since the tests work most of the time, my guess is that the configuration "works", but sometimes gets overloaded with tasks when running in CI. To debug I would need a failure with the docker-compose logs from the fixture environment. 

Potentially `wait_until_valid` could just wait longer and the task queue would drain eventually. Making the `DJANGO_DANDI_VALIDATION_JOB_INTERVAL` longer would actually reduce the number of tasks the system has to consume, since each check is it's own task that needs to be run.

**Comment by yarikoptic** [comment]
@dchiquito what do you think should be a reasonable (should never happen) cut off to take instead of 20 seconds?  60?

**Comment by dchiquito** [comment]
20 seconds is reasonable, and most of the time it works fine. When it does fail, it seems to fail all tests that check for validation, never just one or two. That seems to me like something is going wrong with the docker-compose set up, but I can't tell for sure without docker-compose logs. Maybe cranking it up to 60 will help, maybe not.

**Comment by waxlamp** [comment]
@dchiquito, @jwodder: is this bug still manifesting?

**Comment by dchiquito** [comment]
I think this is duplicated by https://github.com/dandi/dandi-api/issues/567

**Comment by waxlamp** [comment]
Since you are not sure, I'll leave this open; but go ahead and close this if you are sure they're the same underlying problem.

**Comment by danlamanna** [comment]
This seems stale and possibly resolved by #567. Closing but feel free to reopen @jwodder.
